syne_tune/blackbox_repository/conversion_scripts/scripts/yahpo_import.py

```diff
@@ -206,11 +207,18 @@ def map_key(k: str) -> str:
         else:
             return config
 
-    def _active_hyperparameters(self, configuration: Dict[str, Any]) -> List[str]:
+    def _get_active_hyperparameters(self, configuration: Dict[str, Any]) -> List[str]:
```

Comment:
It already has `configuration` as input anyway.

Reply:
In fact, even better if the method got passed the value for the fidelity attribute, and then it returns the filtered config with the fidelity attribute. This would leave the code below simple.

Reply:
Okay, the below code is now a lot simpler. We are redoing the calculation for which hyperparameters are active when we don't give in a fidelity value, but it doesn't seem to slow things down much. I renamed `_get_active_hyperparameters()` to `_prepare_yahpo_configuration()` as it's now doing multiple things. I also separated out the code for parsing the fidelity if given into a new function `_parse_fidelity()`